tests/inet/inet.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/sirupsen/logrus"
+)
+
+// This is similar to netcat's listen mode, except it logs which port it's
+// assigned if it's told to attempt to bind to port 0.  Or it's similar to
+// inetd, if it wasn't a daemon, and it wasn't as well-written.
+func main() {
+	pidFile := ""
+	portFile := ""
+	detach := false
+	flag.BoolVar(&detach, "detach", false, "detach from terminal")
+	flag.StringVar(&portFile, "port-file", "", "file to write listening port number")
+	flag.StringVar(&pidFile, "pid-file", "", "file to write process ID to")
+	flag.Parse()
+	args := flag.Args()
+	if len(args) < 1 {
+		fmt.Printf("Usage: %s [-port-file filename] [-pid-file filename] command ...\n", filepath.Base(os.Args[0]))
+		return
+	}
+	// Start listening without specifying a port number.
+	ln, err := net.ListenTCP("tcp", &net.TCPAddr{})
```

Comment:
would it make sense to at least bind 127.0.0.1, there seems little reason to expose this externally

Reply:
Hmm, leaving it empty means we don't have to care about ipv4/ipv6 here, though.

Reply:
I mean fair enough, it should not be a problem though as all test connect to localhost which generally should try both `::1` and `127.0.0.1` so I doubt it causes issues.